docs/PROBLEM-SOLVING.md

```diff
@@ -0,0 +1,42 @@
+We find that often, the syntax or structure of a programming language isn't the most difficult part of solving a challenge - figuring out _what to code_ or _where to start_ is.
+Developing strategies and a system to break down problems into code-able steps is a **great** skill to cultivate.
+Below are some community-sourced articles, videos, and books that can help you develop this super-skill.
+
+
+- Free Code Camp offers a good overview article on [How to Think Like a Programmer - Lessons in Problem-Solving][free-code-camp-think-like-a-programmer].
+- Kurtis Pykes writing for Towards Data Science has a nice [hands-on tutorial for problem-solving][Kurtis Pykes: Hands-on Tutorial - How to Improve Your Problem-Solving Skills as A Programmer].
+- UC Berkeley has a nice PDF summary of [G. Polya's Problem Solving Techniques][g-polya-how-to-solve-it-summary].
+- Originally written in 1945 as guidance for tackling complicated math problems,[G. Polya's How To Solve It][g-polya-how-to-solve-it] (full book) is still _excellent_ advice for problem-solving in general.
+- Mentioned in the Free Code Camp Article, V. Anton Spraul's [Think Like a Programmer: An Introduction to Creative Problem Solving][v-anton-spraul-think-like-a-programmer] is a more modern and programming-focused take on the same general methods Polya outlines for mathematics.
+- [Felienne Hermans][felienne-hermans] is more focused on _how_ people learn the art of coding and how that maps to learning in general.
+  She has written [The Programmers Brian][programmers-brain-free-online], with strategies for reading code better, thinking about code clearly, writing better code, and becoming a better code collaborator.
+  - You can order a physical copy of her book [here][programmers-brain-manning].
+  - She also has two interesting videos on the same topic: [How to Teach Programming (and other things??)][felienne-hermans-how-to-teach-programming] and [Writing is Programming is Writing][felienne-hermans-programming-is-writing-is-programming].
+- [Algorithmic Thinking (Daniel Zingaro)][daniel-zingaro-algorithmic-thinking] from No Starch Press and [How to think like a programmer (Paul Vickers)][paul-vickers-how-to-think-like-a-programmer] also cover problem-solving when coding.
+- [Computational Thinking: A beginner's guide to problem-solving and programming (Karl Beecher)][beecher-computational-thinking] and [Computational Thinking for the Modern Problem-Solver (Riley and Hunt)][riley-and-hunt-computational-thinking]. Also offer strategies for thinking about problems and breaking them down for coding solutions.
+- [From Computing to Computational Thinking (Paul S. Wang)][wang-computational-thinking] is a bit more generalized and academic, but offers problem-solving strategies and structures that can apply to more than coding, and does not require prior programming knowledge.
+
+Many of the books above can also be read through the Association of Computing Machinery library.
```

Comment:
```suggestion
Some of the books above may also be read through the Association of Computing Machinery library.
```

Edited to account for O'Reilly leaving ACM learning resources at the end of June. Skillsoft, which remains with ACM, does have at least one of these books last I checked.

Reply:
So frustrating about O'Reilly.  My dream is that we talk them into some limited access as part of maybe an in-kind donation.  Like we select 100 (or some such) books that students can access for resources if they are exercism members.  Or something.  But for now... _sigh_.

Reply:
Yeah, I wrote an email to them about that. They justified the higher fee because they've added a lot of content besides books. I suggested offering just books for a reduced fee, because I wasn't interested in the other content. They never responded.